Design/SANS/BeamCentreFinderDesign.md

```diff
@@ -0,0 +1,119 @@
+# Design Document for Beam Centre Finder
+
+### Current issue
+Currently the Beam Centre Finder(BCF) is heavily tied up with the ISIS instruments and the ReductionSingleton().
+ANSTO are looking currently for similar functionality but since their reduction workflow is not based on the
+ReductionSingleton() and they don't make use of the instrument class they cannot use the BCF directly. 
+This document lines out the required changes to make the BCF useable in a more general context which could 
+also be beneficial when moving away from the current Reduction-Step approach.
+
+### Proposed solution
+In general terms we want to hide any solution-specific details in the BCF. Currently there is already a layer of 
+abstration which hides the instrument specific details, e.g. LARMOR uses an instrument rotation instead of a translation.
+A common generic interface should allow for the current implementation on the ISIS side as well as other 
+implementations to make use of the Beam Centre Finder logic.
```

Comment:
We should base the design on the current geometry. Since there is no way for an instrument component to know it's degrees of freedom. Individual implementations should have a **simple** way to specify how the banks could be moved. We should only deal with current cases of translation and rotation (one axis)